doc/source/whatsnew/v1.0.1.rst

```diff
@@ -24,6 +24,7 @@ Deprecations
 Bug fixes
 ~~~~~~~~~
 - Bug in :meth:`GroupBy.apply` was raising ``TypeError`` if called with function which returned a non-pandas non-scalar object (e.g. a list) (:issue:`31441`)
+- Bug in :meth:`GroupBy.apply` where the output index type was depending on internals (:issue:`31612`)
```

Comment:
GroupBy isn't in the top level. Usually we write this as \`\`.groupby().apply()\`\`.

And as a user, I wouldn't know what to make of "depending on the internals".

Reply:
Perhaps something like "where the result shape was incorrect when the output index was not identical to the input index"?

Reply:
Changed it to :meth:`core.groupby.DataFrameGroupBy.apply` which should be rendered properly by sphinx